ceph-deploy-pull-requests/build/setup

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -ex
+
+# Create the virtualenv
+virtualenv venv
+. venv/bin/activate
+
+# Define and ensure the PIP cache
+PIP_SDIST_INDEX="$HOME/.cache/pip"
+mkdir -p $PIP_SDIST_INDEX
+
+# Define and ensure a GIT Ansible cache
+REPO_CACHE="$HOME/.cache/git/ceph-build"
+mkdir -p $REPO_CACHE
+
+# Clone the ceph-build repo or pull if it exists
+if [ -d "$REPO_CACHE/.git" ]; then
+    cd "$REPO_CACHE" && git pull
+    cd -
+else
+    git clone https://github.com/ceph/ceph-build.git "$REPO_CACHE"
```

Comment:
wait, I don't think that using ceph-build as a second remote would make sense for setting up. This should only run when the job runs, so it properly sets it up. If you set up ceph-build as a second remote this setup will run but it may run on any host not necessarily where/when we need it